eclair-core/src/test/scala/fr/acinq/eclair/db/ChannelsDbSpec.scala

```diff
@@ -324,7 +324,20 @@ class ChannelsDbSpec extends AnyFunSuite {
         }
       }
     )
+  }
 
+  test("json column reset (postgres)") {
+    val dbs = TestPgDatabases()
+    val db = dbs.channels
+    val channel = ChannelCodecsSpec.normal
+    db.addOrUpdateChannel(channel)
+    dbs.connection.execSQLUpdate("UPDATE local_channels SET json='{}'")
+    db.asInstanceOf[PgChannelsDb].resetJsonColumns(dbs.connection)
+    assert({
+      val res = dbs.connection.execSQLQuery("SELECT * FROM local_channels")
+      res.next()
+      res.getString("json").length > 100
```

Comment:
Can't we test equality properly?

Reply:
It's not straightforward because with the `JSONB` type, Postgres mutates the data:

> The json data type stores an exact copy of the input text, which processing functions must reparse on each execution; while jsonb data is stored in a decomposed binary format that makes it slightly slower to input due to added conversion overhead, but significantly faster to process, since no reparsing is needed

https://www.postgresql.org/docs/13/datatype-json.html

In theory we could deserialize then reserialize but... we don't support deserialization. I think it's okay if we assume that this test is only meant to test that the data is actually reset?

Reply:
Yes, that sounds good enough!